### PR TITLE
Fully remove pytest dependency

### PR DESCRIPTION
**Pull request by patrickportal**
Replace `pytest` and `pytest-cov` by `unittest` and `coverage`
This PR is first work of @chris-verclytte and contains a bit of https://github.com/ChauffeurPrive/nestor-api/pull/40 (which will be closed soon)

### CODE REVIEW

File: tests/config_validator/test_validate.py [hunk 1]
```diff
@@ -1,111 +1,114 @@
+# pylint: disable=bad-continuation
```
Comment (patrickportal) [inline tests/config_validator/test_validate.py:1]: This file has already been reviewed in https://github.com/ChauffeurPrive/nestor-api/pull/41/files

File: tests/api/test_wsgi.py [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+from unittest.mock import patch
+
+
+class TestApi(TestCase):
+    @patch("nestor_api.api.flask_app.create_app", autospec=True)
+    def test_api_instance(self, create_app_mock):
+        """Should instance API"""
+        import nestor_api.api.wsgi  # pylint: disable=import-outside-toplevel,unused-import
+
+        create_app_mock.assert_called()
```
Comment (patrickportal) [inline tests/api/test_wsgi.py:12]: I have no idea why coverage was 100% without this when we used pylint 🤷‍♂️

File: tests/api/test_wsgi.py [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
```
Comment (patrickportal) [inline tests/api/test_wsgi.py:11]: I have no idea why coverage was 100% without this with pytest-cov 🤷‍♂️

File: tests/lib/test_io.py [hunk 4]
```diff
@@ -123,6 +123,15 @@ def test_get_temporary_directory_path(self, get_working_path_mock):
...
 
+    @patch("nestor_api.lib.io.get_working_path", autospec=True)
+    def test_get_temporary_directory_path_default(self, get_working_path_mock):
+        get_working_path_mock.return_value = "/tmp/nestor/work/1594903736050223-454134888"
+        generated_path = io.get_temporary_directory_path()
+
+        tmp_directory_name = get_working_path_mock.call_args[0][0]
+        self.assertRegex(tmp_directory_name, "^[0-9]{14,16}-[0-9]{9}$")
+        self.assertEqual(generated_path, "/tmp/nestor/work/1594903736050223-454134888")
+
```
Comment (patrickportal) [inline tests/lib/test_io.py:134]: New lib `coverage` knows when a `if` statement is never false and won't say the if is cover when not a true + a false value, which is good 👍

File: tests/api/api_routes/builds/test_build_app.py [hunk 5]
```diff
@@ -141,6 +141,29 @@ def test_build_app_handle_errors(
             "[/api/builds/:app] Error while tagging and building the app",
         )
 
+    def test_build_app_handle_early_errors(
```
Comment (patrickportal) [inline tests/api/api_routes/builds/test_build_app.py:144]: Same as https://github.com/ChauffeurPrive/nestor-api/pull/41/files#r455779902, new `coverage` lib needs to has `if` called with true and false

File: tests/api/test_wsgi.py [hunk 6]
```diff
@@ -0,0 +1,11 @@
+from unittest import TestCase
+from unittest.mock import patch
+
+
+class TestApi(TestCase):
+    @patch("nestor_api.api.flask_app.create_app", autospec=True)
+    def test_api_instance(self, create_app_mock):
+        """Should instance API"""
```
Comment (benard-g) [inline tests/api/test_wsgi.py:8]: You normally don't have to write a docstring.
```suggestion
```
```suggestion
        """Should instantiate API"""
```

File: tests/config_validator/test_validate.py [hunk 7]
```diff
@@ -1,111 +1,114 @@
...
+        self, get_validation_target_mock, build_apps_path_mock
+    ):
+        fixtures_path = Path(os.path.dirname(__file__), "..", "__fixtures__").resolve()
+        build_apps_path_mock.return_value = fixtures_path
+        get_validation_target_mock.return_value = "SOME_VALUE"
+
+        with self.assertRaisesRegex(
+            Exception,  # pylint: disable=bad-continuation
+            "There is no configuration to be validated. Be sure to define a valid NESTOR_VALIDATION_TARGET",  # pylint: disable=bad-continuation,line-too-long
+        ):
```
Comment (benard-g) [inline tests/config_validator/test_validate.py:82]: Why the rollback here ? (I found it clearer without the comments on every line)

File: tests/config_validator/test_validate.py (hunk 1 above)
Comment (patrickportal) [reply tests/config_validator/test_validate.py:1]: I meant https://github.com/ChauffeurPrive/nestor-api/pull/40

File: tests/config_validator/test_validate.py (hunk 7 above)
Comment (patrickportal) [reply tests/config_validator/test_validate.py:82]: _Could be because of  `assertRaisesRegex` but not sure what you meant here_

**Pull request merged**